corehq/motech/repeaters/models.py

```diff
@@ -248,8 +325,291 @@ def reset_next_attempt(self):
+        return self.connection_settings.url.format(domain=new_domain)
+
+    def send_request(self, repeat_record, payload):
+        """Add custom response handling to allow more nuanced handling of form errors"""
+        return get_repeater_response_from_submission_response(
+            super().send_request(repeat_record, payload)
+        )
+
+    @classmethod
+    def _migration_get_couch_model_class(cls):
+        return ReferCaseRepeater
+
+
+class SQLDataRegistryCaseUpdateRepeater(SQLCreateCaseRepeater):
+    """
+    A repeater that triggers off case creation but sends a form to update cases in
+    another commcare project space.
+    """
+    class Meta:
+        proxy = True
+
+    friendly_name = _("Update Cases in another CommCare Project via a Data Registry")
+    payload_generator_classes = (DataRegistryCaseUpdatePayloadGenerator,)
+
+    def form_class_name(self):
+        return 'DataRegistryCaseUpdateRepeater'
+
+    @classmethod
+    def available_for_domain(cls, domain):
+        return toggles.DATA_REGISTRY_CASE_UPDATE_REPEATER.enabled(domain)
+
+    def get_url(self, repeat_record):
+        new_domain = self.payload_doc(repeat_record).get_case_property('target_domain')
+        return self.connection_settings.url.format(domain=new_domain)
+
+    def send_request(self, repeat_record, payload):
+        return get_repeater_response_from_submission_response(
+            super().send_request(repeat_record, payload)
+        )
+
+    def allowed_to_forward(self, payload):
+        if not super().allowed_to_forward(payload):
+            return False
+
+        # Exclude extension cases where the host is also a case type that this repeater
+        # would act on since they get forwarded along with their host
+        host_index = payload.get_index(CASE_INDEX_IDENTIFIER_HOST)
+        return not host_index or host_index.referenced_type not in self.white_listed_case_types
+
+    @classmethod
+    def _migration_get_couch_model_class(cls):
+        return DataRegistryCaseUpdateRepeater
+
+
+class SQLShortFormRepeater(SQLRepeater):
+    """
+    Record that form id & case ids should be repeated to a new url
+
+    """
+    FORMAT_OPTIONS = ['short_form_json']
+
+    format = OptionValue(choices=FORMAT_OPTIONS)
+
+    class Meta:
+        proxy = True
+
+    version = OptionValue(choices=LEGAL_VERSIONS, default=V2)
+    friendly_name = _("Forward Form Stubs")
+
+    payload_generator_classes = (ShortFormRepeaterJsonPayloadGenerator,)
+
+    @memoized
+    def payload_doc(self, repeat_record):
+        return FormAccessors(repeat_record.domain).get_form(repeat_record.payload_id)
+
+    def allowed_to_forward(self, payload):
+        return payload.xmlns != DEVICE_LOG_XMLNS
+
+    def get_headers(self, repeat_record):
+        headers = super(ShortFormRepeater, self).get_headers(repeat_record)
+        headers.update({
+            "received-on": self.payload_doc(repeat_record).received_on.isoformat()+"Z"
```

Comment:
I'm assuming this code was copied from somewhere else, but this is not a good way to add a timezone to a datetime. For example, if it already has a timezone then this will create malformed output.

Reply:
I think calling `.isoformat()` would take care of the `datetime` object already having a timezone. So we would always get a string that does not have `Z` in the end.
But would be definitely interested in knowing if you have some suggestions on how to get it done?

Reply:
I did some reading about what valid ISO formats are. I understood your point. 
`2022-01-10T18:00:52.609500Z` would be a valid ISO Date but `2022-01-10T15:02:35.312010+05:53Z` would be not. 
So should we check if the datetime is having timezone info in it and ignore the ones that already have one?

Reply:
We have a tool for this:
```python
from dimagi.utils.parsing import json_format_datetime
```
...
```suggestion
            "received-on": json_format_datetime(self.payload_doc(repeat_record).received_on)
```
`received_on.tzinfo` should be `None` cos that's how we create it, but `json_format_datetime()` checks.

Reply:
`json_format_datetime()` will change behavious though! The string that it returns includes microseconds, where `isoformat()` returns seconds (if called with its default params).

Reply:
No, you're right. I think I missed something. I must have been playing with `isoformat()` using an object that didn't have microseconds.

Reply:
Thank you both for your suggestions 🙏